src/credentials/FabricTable.cpp

```diff
@@ -1248,97 +1383,548 @@ CHIP_ERROR FabricTable::AllocatePendingOperationalKey(Optional<FabricIndex> fabr
+        VerifyOrDie(liveFabricEntry != nullptr);
+
+        CHIP_ERROR metadataErr = StoreFabricMetadata(liveFabricEntry);
+        if (metadataErr != CHIP_NO_ERROR)
+        {
+            ChipLogError(FabricProvisioning, "Failed to commit pending fabric metadata: %" CHIP_ERROR_FORMAT, metadataErr.Format());
+        }
+        stickyError = (stickyError != CHIP_NO_ERROR) ? stickyError : metadataErr;
+
+        // We can only manage commissionable pending fail-safe state if we have a keystore
+        CHIP_ERROR keyErr = CHIP_NO_ERROR;
+        if ((mOperationalKeystore != nullptr) && mOperationalKeystore->HasOpKeypairForFabric(fabricIndex) &&
+            mOperationalKeystore->HasPendingOpKeypair())
+        {
+            keyErr = mOperationalKeystore->CommitOpKeypairForFabric(fabricIndex);
+            if (keyErr != CHIP_NO_ERROR)
+            {
+                ChipLogError(FabricProvisioning, "Failed to commit pending operational keypair %" CHIP_ERROR_FORMAT,
+                             keyErr.Format());
+                mOperationalKeystore->RevertPendingKeypair();
+            }
+        }
+        stickyError = (stickyError != CHIP_NO_ERROR) ? stickyError : keyErr;
+
+        // Commit operational certs
+        CHIP_ERROR opCertErr = mOpCertStore->CommitOpCertsForFabric(fabricIndex);
+        if (opCertErr != CHIP_NO_ERROR)
+        {
+            ChipLogError(FabricProvisioning, "Failed to commit pending operational certificates %" CHIP_ERROR_FORMAT,
+                         opCertErr.Format());
+            mOpCertStore->RevertPendingOpCerts();
+        }
+        stickyError = (stickyError != CHIP_NO_ERROR) ? stickyError : opCertErr;
+
+        // Failure to commit Last Known Good Time is non-fatal.  If Last Known
+        // Good Time is incorrect and this causes incoming certificates to
+        // appear invalid, the certificate validity policy will see this
+        // condition and can act appropriately.
+        CHIP_ERROR lkgtErr = mLastKnownGoodTime.CommitPendingLastKnownGoodChipEpochTime();
+        if (lkgtErr != CHIP_NO_ERROR)
+        {
+            // Log but this is not sticky...
+            ChipLogError(FabricProvisioning, "Failed to commit Last Known Good Time: %" CHIP_ERROR_FORMAT, lkgtErr.Format());
+        }
+
+        // If an Add occurred, let's update the fabric index
+        CHIP_ERROR fabricIndexErr = CHIP_NO_ERROR;
+        if (mStateFlags.Has(StateFlags::kIsAddPending))
+        {
+            UpdateNextAvailableFabricIndex();
+            fabricIndexErr = StoreFabricIndexInfo();
+            if (fabricIndexErr != CHIP_NO_ERROR)
+            {
+                ChipLogError(FabricProvisioning, "Failed to commit pending fabric indices: %" CHIP_ERROR_FORMAT,
+                             fabricIndexErr.Format());
+            }
+        }
+        stickyError = (stickyError != CHIP_NO_ERROR) ? stickyError : fabricIndexErr;
+    }
+
+    FabricIndex previouslyPendingFabricIndex = fabricIndex;
```

Comment:
Why do we need this extra variable?  Nothing changes `fabricIndex` between here and the one use of `previouslyPendingFabricIndex`.